modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase

```diff
@@ -655,4 +655,189 @@ definition {
+				userEmailAddress = "supplieruser@liferay.com",
+				userFirstName = "Supplier",
+				userLastName = "User",
+				userRole = "Account Supplier",
+				userScreenName = "supplieruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And a product belongs to another catalog is created") {
+			CommerceJSONProductsAPI._addCommerceProduct(
+				catalogName = "Master",
+				productName = "Simple Product",
+				productType = "Simple",
+				sku = "SKU1001");
+		}
+
+		task ("And Supplier account is linked to Minium catalog") {
+			CommerceCatalog.openCatalogsAdmin();
+
+			CommerceNavigator.gotoEntry(entryName = "Minium");
+
+			Click(
+				key_fieldFieldLabel = "Link Catalog to a Supplier",
+				locator1 = "TextInput#SELECTOR_INPUT",
+				value1 = "Commerce Supplier");
+
+			Click(
+				key_categoryFilter = "Commerce Supplier",
+				locator1 = "CommerceEntry#FILTER_SELECT_CATEGORY");
+
+			AssertTextEquals(
+				key_fieldFieldLabel = "Link Catalog to a Supplier",
+				locator1 = "TextInput#SELECTOR_INPUT",
+				value1 = "Commerce Supplier");
+
+			Button.clickSave();
+		}
+
+		task ("And one discount for the price product 'ABS Sensor' and one for the promotion product 'Brake Pads' are created by admin") {
+			CommerceJSONDiscountsAPI._addDiscount(
+				active = "true",
+				level = "L1",
+				percentageLevel1 = 10,
+				productNames = "ABS Sensor,Brake Pads",
+				target = "products",
+				title = "Test Discount",
+				usePercentage = "true");
+		}
+
+		task ("When the supplier user logs in") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "supplieruser@liferay.com",
+				userLoginFullName = "Supplier User");
+		}
+
+		task ("Then the supplier user cannot view the Discounts panel") {
+			ApplicationsMenuHelper.openApplicationsMenu();
+
+			ApplicationsMenu.gotoPanel(panel = "Commerce");
+
+			AssertElementNotPresent(
+				key_category = "Pricing",
+				key_portlet = "Discounts",
+				locator1 = "ApplicationsMenu#PORTLET");
+		}
+
+		task ("And the supplier user can view the two products with discounts that admin added before") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/catalog");
+
+			CommerceAcceleratorsInitializer.changeProductsSorting(sortingType = "Name Ascending");
+
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "ABS Sensor");
+
+			CommerceAcceleratorsInitializer.viewDiscountedProductDetailsPage(
+				discountPercentage = 10,
+				productDiscountedPrice = "45.00",
+				productListPrice = "50.00");
+
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/catalog");
+
+			CommerceAcceleratorsInitializer.changeProductsSorting(sortingType = "Name Ascending");
+
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "Brake Pads");
+
+			CommerceAcceleratorsInitializer.viewDiscountedProductDetailsPage(
+				discountPercentage = 10,
+				productDiscountedPrice = "17.01",
+				productListPrice = "21.00",
+				productPromoPrice = "18.90");
+		}
```

Comment:
The supplier shouldn't be the account to check the store front. Use a business account with a buyer user instead.